web/src/lib/utils/helpers.test.ts

```diff
@@ -0,0 +1,43 @@
+import { useMountEffectWhenDefined } from "@/lib/utils/helpers";
+import { renderHook } from "@testing-library/react";
+
+describe("useMountEffectWhenDefined", () => {
+  interface Props {
+    value: string | undefined;
+  }
+
+  it("calls function once when single dependency becomes defined", () => {
+    const mockFn = jest.fn();
+    const { rerender } = renderHook(
+      (props: { value: string | undefined }) => {
+        useMountEffectWhenDefined(mockFn, props.value);
+      },
+      {
+        initialProps: { value: undefined } as Props,
+      }
+    );
+
+    expect(mockFn).not.toHaveBeenCalled();
+
+    rerender({ value: "defined" });
+    expect(mockFn).toHaveBeenCalledTimes(1);
+
+    rerender({ value: "changed" });
+    expect(mockFn).toHaveBeenCalledTimes(1);
+  });
```

Comment:
[dust] Should there also be a version of this where `initialProps` is defined from the beginning?

Reply:
good call, updated these tests to include that scenario